content/articles/building-web-application-with-kotlin-and-typescript/index.md

```diff
@@ -0,0 +1,191 @@
+
+- yarnInstall: A call to yarn installs the expected JS modules.
+
+- UnpackKotlinJs: JS files are unpacked from nodekotlin dependencies.
+
+- nodebuild: Angular/node.js build gets invoked.
+
+### Using Kotlin generated modules in TypeScript code
+
+Now that the Kotlin-JS modules are in the node_modules directory we can create the Angular/TypeScript code. There are a few other ways to do it but we will consider using the import statement method since the javascript module does most of the work.
+
+Following is a way of importing generated Kotlin modules to a TypeScript file:
+
+```
+import * as info_js from 'example.addressbook-information';
+import info = info_js.example.addressbook.information;
+...
+let contact = new info.contact('alias')
+```
+import statement on the second time is not much important it also develops a good alias for the content of the modules which helps to reduce more writing of fully qualified names all the time we use Kotlin generated modules.
+
+### TypeScript types for Kotlin code
+
+To generate a *.d.ts file containing the same number of Typescript declarations, we can use a chunk of Kotlin reflection and kotlin string templates to cross the Kotlin classes out of the information module. It becomes possible in the build if we include a JVM target.
+
+A package.json file must be created containing a 'types' entry, which should refer to the type declaration generated and the generated JavaScript code as illustrated below:
+
+```
+{
+  "name": "com.example-my-module-common"
+  "version": "1.0.0",
+  "main": "./com.example-my-module-common.js",
+  "types": "./com.example-my-module-common.d.ts"
+}
+```
+
+### TypeScript declarations for third party libraries
+
+The same JVM reflection approach is used, provided that the third party is a Kotlin multiplatform library that includes a JVM target. JVM divides the Kotlin stdlib into several modules while the javascript module provided by Kotlin contains everything thus it becomes crucial to develop its typescript declaration.  
+
+According to the Gradle, the naming conversion is ``` <group>:<name>:<version>``` that is accepted worldwide. Unfortunately, names for both JVM and JavaSript modules do not always match. Thus, it becomes necessary to have code generators for the Typescript declaration that helps in mapping between the Kotlin module and the JVM module. The generators play an important role of correctly adding the imports file to the Typescript declaration. 
+
+### Conclusion
+
+Integration of Kotlin generated modules with Typescript web browser application has been well illustrated in this article. It requires build script codes which are simplified by the use Gradle plugin to set up the Kotlin multiplartform build. After unpacking the KotlinJs modules , it becomes easier to use the Kotlin generated modules into TypeScrits codes of a web application.

```

Comment:
Simplify "It requires build script codes which are simplified by the use Gradle plugin..." to "It requires build script codes simplified by the Gradle plugin…".
It should be "multiplatform" not "multiplartform"